frontend/catalyst/compiler.py

```diff
@@ -408,6 +418,7 @@ def get_default_flags(options):
             f"-l{openblas_lib_name}",  # required for custom_calls lib
             "-lcustom_calls",
             "-lmlir_async_runtime",
+            "-lpython" + version_str,
```

Comment:
Could you explain why we are linking in python? :)

Reply:
Its required by the runtime CAPI. For non-conda environment, we do not need that. However, conda will install its own python-dev library.

Reply:
Oh right, I forgot that in the meantime we have created a core dependency on Python in the runtime, not just in some device backends.